server/services/trade.ts

```diff
@@ -80,6 +89,13 @@ export default class TradeService extends EventEmitter {
             throw new ValidationError(`Cannot send credits to yourself.`);
         }
 
+        if (fromPlayer.userId && this.isTradingAllyRestricted(game)) {
```

Comment:
I don't think you need to check if the `userId` is set here. Might want to check if formal alliances is enabled though.

Reply:
Will remove the extra check for player ids.  The formal alliance check is done in isTradingAllyRestricted()